File: 프로그래머스/lv2/42587. 프린터/프린터.py
def solution(priorities, location):
    new_priorities = []
    answer_priorities = []
    max_priority = max(priorities)
    
    for idx, i in enumerate(priorities):
        new_priorities.append([idx, i])

    while len(new_priorities) != 0:
        max_priority = max(priorities)
            
        if new_priorities[0][1] == max_priority:
            answer_priorities.append(new_priorities[0])
            new_priorities.pop(0)
            priorities.remove(max(priorities))

        else:
            a = new_priorities.pop(0)
            new_priorities.append(a)
            max_priority = max(priorities)
            
            
    return [idx+1 for idx, content in enumerate(answer_priorities) if content[0] == location][0]
    

File: 프로그래머스/lv2/42587. 프린터/프린터.js
function solution(priorities, location) {
    const newPriorities = new Array(priorities.length).fill().map((_,idx) => [priorities[idx],idx] );
    const result = [];
    let answer = 0;

    while(priorities.length !== 0){
        let first = priorities.shift();
        let newFirst = newPriorities.shift();
        let max = Math.max(...priorities);

        if(first < max){
            newPriorities.push(newFirst);
            priorities.push(first);
        }else{
            result.push(newFirst);
        }
    }

    result.forEach((el,idx) => {
        if(el[1] === location){
            answer = idx+1;
        };
    })

    return answer;
}